external-builds/pytorch/repo_management.py

```diff
@@ -217,11 +217,36 @@ def commit_hipify(args: argparse.Namespace):
     # the root repo first, it will not add submodule changes.
     all_paths = get_all_repositories(repo_dir)
     for module_path in all_paths:
+        # 1) Quick pre-check
         status = list_status(module_path)
         if not status:
+            print(f"[hipify] No changes in {module_path}; skipping commit")
             continue
-        print(f"HIPIFY made changes to {module_path}: Committing")
+
+        print(f"HIPIFY made changes to {module_path}: preparing commit")
+
+        # 2) Stage everything HIPIFY may have produced
         exec(["git", "add", "-A"], cwd=module_path)
+
+        # 3) Only commit if the *staging area* has changes.
+        #    `git diff --cached --quiet` returns:
+        #       0  -> no staged changes
+        #       1  -> staged changes exist
+        try:
+            subprocess.check_call(
+                ["git", "diff", "--cached", "--quiet"],
+                cwd=str(module_path),
+                stdout=subprocess.DEVNULL,
+                stderr=subprocess.DEVNULL,
+            )
+            # Exit code 0: nothing staged -> skip committing to avoid exit 1
+            print(f"[hipify] Nothing staged in {module_path}; skipping commit")
+            continue
```

Comment:
I would expect to see this in the logs if this code is fixing the issue, but I don't see it on any of the jobs from https://github.com/ROCm/TheRock/actions/runs/17108368569. Can you re-run enough times to be sure that this fixes the issues?

I'm also wondering if something transient about repository state is impacting the checkout step. We may need https://github.com/ROCm/TheRock/issues/1236 solved to fix this with confidence.